packages/xc-sidebar/src/utils/getStyles.ts

```diff
@@ -1,31 +1,311 @@
+  appearance = 'default',
+  baseAppearance = 'default',
+  defaultPropertyValue = 'inherit',
+) => {
+  let property = sidebarAppearanceTheme(theme, appearance, baseAppearance, [propertyPath]);
+
+  if (!property) {
+    property = defaultPropertyValue;
+  }
+
+  return memoizee((
+    elementName,
+  ) => {
+    const element = sidebarAppearanceTheme(theme, appearance, baseAppearance, elementName);
+
+    return css`
+      ${() => `${propertyPath}: ${get(element, [propertyPath], property)}`};
+    `;
+  });
+});
+
+export const getWidthStyles: IReturnFunctionForConcreteProp<
+IReturnWithArgsFunction<any, FlattenSimpleInterpolation>
+> = memoizee((
+  theme,
+  appearance = 'default',
+  baseAppearance = 'default',
+) => {
+  let width = sidebarAppearanceTheme(theme, appearance, baseAppearance, ['width']);
+
+  if (!width) {
+    width = 'auto';
+  }
+
+  return memoizee((
+    elementName,
+    shouldFitContainer,
+  ) => {
+    const element = sidebarAppearanceTheme(theme, appearance, baseAppearance, elementName);
+
+    return css`
+      width: ${shouldFitContainer ? '100%' : get(element, ['width'], width)};
+    `;
+  });
+});
+
+export const getBackgroundColorStyles: IReturnFunctionForConcreteProp<
+IReturnWithArgsFunction<any, FlattenSimpleInterpolation>
+> = memoizee((
+  theme,
+  appearance = 'default',
+  baseAppearance = 'default',
+) => {
+  let backgroundColor = sidebarAppearanceTheme(theme, appearance, baseAppearance, ['backgroundColor']);
+
+  if (!backgroundColor) {
+    backgroundColor = 'transparent';
+  }
+
+  return memoizee((
+    elementName,
+  ) => {
+    const element = sidebarAppearanceTheme(theme, appearance, baseAppearance, elementName);
+
+    return css`
+      background-color: ${get(element, ['backgroundColor'], backgroundColor)};
+    `;
+  });
+});
+
+export const getZIndexStyles: IReturnFunctionForConcreteProp<
+IReturnWithArgsFunction<any, FlattenSimpleInterpolation>
+> = memoizee((
+  theme,
+  appearance = 'default',
+  baseAppearance = 'default',
+) => {
+  let zIndex = sidebarAppearanceTheme(theme, appearance, baseAppearance, ['zIndex']);
+
+  if (!zIndex) {
+    zIndex = 'unset';
+  }
+
+  return memoizee((
+    elementName,
+  ) => {
+    const element = sidebarAppearanceTheme(theme, appearance, baseAppearance, elementName);
+
+    return css`
+      z-index: ${get(element, ['zIndex'], zIndex)};
+    `;
+  });
+});
+
+export const getPaddingStyles: IReturnFunctionForConcreteProp<
+IReturnWithArgsFunction<any, FlattenSimpleInterpolation>
+> = memoizee((
+  theme,
+  appearance = 'default',
+  baseAppearance = 'default',
+) => {
+  let commonPadding = sidebarAppearanceTheme(theme, appearance, baseAppearance, ['padding']);
+
+  if (!commonPadding) {
+    commonPadding = 0;
+  }
+
+  return memoizee((elementName) => {
+    const elementPadding = sidebarAppearanceTheme(theme, appearance, baseAppearance, [elementName, 'padding']);
+    const padding = !isNil(elementPadding) ? elementPadding : commonPadding;
```

Comment:
There is one problem if I will set `padding` like `10px 10px` your construction will not work. 

It would be correct be if you will return `padding` and padding side together. Also, if a prop is not set you don't have to return it